### DISCUSSION TITLE
auto quant generate encoding json, cannot apply to qairt 2.32&&2.31

### DISCUSSION THREAD

**Discussion by lgyStoic** [quic/aimet#3975] [Q&A] [answered]
I Use AIMET AutoQuant function to quant my model successfully, but when I apply to SNPE 
will raise error
![image](https://github.com/user-attachments/assets/0d558189-229b-4eae-b25f-6bff35faa9a1)
Though error I guess the json encoding json file is not compatible to this version SNPE ? cause I found the per-channel description is totally different.
in AIMET generate json file
![image](https://github.com/user-attachments/assets/1c0fe323-b08a-47b8-833d-7528113a10de)
and in SNPE tutorial.
![image](https://github.com/user-attachments/assets/417a2bdf-592d-41a3-ba4e-287f54bbb5f3)
the two format is not the same.  Any one can help?

**Comment by quic-mtuttle** [comment, marked as answer]
Hi @lgyStoic , the example from the SNPE tutorial is using an older encoding format. You can try exporting with this older format to see if that solves the issue you're seeing. To do that, just insert this before calling `sim.export`:

```python
import aimet_common.quantsim
aimet_common.quantsim.encoding_version = '0.6.1'
```

**Reply by lgyStoic** [reply]
Thank you it works !!

**Reply by lgyStoic** [reply]
@quic-mtuttle another question, I use QuantizationSimModel get accuracy is 0.87, but when I apply to SNPE. I using snpe-onnx-dlc  with the encoding json using this "--quantization_overrides", and when quant this model with "--override_params".. But finally this quantized model accuracy only 0.34.. Is there something wrong here? My quantsim code snippet is
```
calibration_data_loader = DataLoader(calibration_dataset, batch_size=BATCH_SIZE)
eval_data_loader = DataLoader(eval_dataset, batch_size=BATCH_SIZE)
dummy_input = torch.randn(1, 3, 420, 560)

from aimet_common.defs import QuantScheme
from aimet_common.quantsim_config.utils import get_path_for_per_channel_config
from aimet_torch.quantsim import QuantizationSimModel
sim = QuantizationSimModel(model,
                           dummy_input=dummy_input,
                           quant_scheme=QuantScheme.post_training_percentile,
                           default_param_bw=8,
                           default_output_bw=8,
                           config_file="/root/workspace/Depth-Anything-V2/htp_quantsim_config_v75.json")
sim.compute_encodings(eval, forward_pass_callback_args=calibration_data_loader)

accuracy = eval(sim.model, eval_data_loader)
print(f'- sim pro accuracy: {accuracy}')
import aimet_common.quantsim
aimet_common.quantsim.encoding_version = '0.6.1'
sim.export(path="./output_quant", filename_prefix="depth_anything_v2_v75", dummy_input=dummy_input)

**Reply by quic-mtuttle** [reply]
Hi @lgyStoic, could you check whether the model you're using passes the [model validator](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/model_validator.html#apiref-torch-model-validator)? 
```python
from aimet_torch.model_validator.model_validator import ModelValidator
ModelValidator.validate_model(model, model_input=dummy_input))
```

This will check whether all the activations in the model are being properly quantized. If not, then you may need to adjust the model definition slightly or pass it through [model preparer](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/model_preparer.html#apiref-torch-model-preparer).

**Reply by lgyStoic** [reply]
> Hi @lgyStoic, could you check whether the model you're using passes the [model validator](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/model_validator.html#apiref-torch-model-validator)?
> 
> ```python
> from aimet_torch.model_validator.model_validator import ModelValidator
> ModelValidator.validate_model(model, model_input=dummy_input))
> ```
> 
> This will check whether all the activations in the model are being properly quantized. If not, then you may need to adjust the model definition slightly or pass it through [model preparer](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/model_preparer.html#apiref-torch-model-preparer).

@quic-mtuttle  That's right, I forget model_preparer, but there another problem in AMP.
I complished in Phase1 
![image](https://github.com/user-attachments/assets/12d78765-6ca8-4008-a066-1bfef2d9afb1)
Phase2, generate a Pareto list
![image](https://github.com/user-attachments/assets/0b3d6192-7f74-44e2-9c2f-c9536519fafd)
but after this I get an error
![image](https://github.com/user-attachments/assets/9b2aeea1-895c-4f1e-8bc3-42a6a13472c8)
aimet can't get ops dim, because of noneType, how to solve this problem?

**Reply by quic-ykota** [reply]
Hello @lgyStoic, to temporarily get past the above error, you could set the flag GreedyMixedPrecisionAlgo.ENABLE_CONVERT_OP_REDUCTION to False (This would disable Phase-3 of AMP). But to fix the issue, please share a reproducible code and we will provide a fix for it.
